src/yarp.c

```diff
@@ -1967,11 +1970,11 @@ parse_targets(yp_parser_t *parser, binding_power_t binding_power, const char *me
   return multi_target;
 }
 
-// Parse a list of statements separated by newlines or semicolons.
-static yp_node_t *
-parse_statements(yp_parser_t *parser, yp_context_t context) {
+// Parse statements separated by newlines or semicolons, and append them to
+// the given list of statements.
+static void
+parse_and_append_statements(yp_parser_t *parser, yp_context_t context, yp_node_t *statements) {
```

Comment:
Not sure I understand the need for this change. Were you forced to create the statements early?